### DISCUSSION TITLE
laragon downloads file instead of serve project

### DISCUSSION THREAD

**Discussion by alzahraa2023** [leokhoa/laragon#897] [General]
in laragon interface when i press on menu->www->project_name to launch the project.test it goes to download a file and not starts to serve the project

**Comment by mustainal** [comment]
up sir, this my problem too

**Reply by leokhoa** [reply]
@mustainal : Can you please provide more details? A screen recorder will be helpful.

**Reply by AWqxKAWERbXo** [reply]
Can't reproduce, too.

**Reply by mustainal** [reply]
so i want to use virtual host laragon on my project before it i using local host but when i open let say "myproject.test" when open on my chrome the tab showing like you want to download some file, but on my new project laravel it is normal can open virtualhost like a normal

**Reply by DigitalBuddhaInc** [reply]
On some of my websites - I get the same issue of trying to view a website with Laragon running... but instead of website loading and being displayed -  a file with short weird name is saved to my downloads folder instead.
For me, the problem seemed to be a mismatch between the version of PHP Laragon was running by default (PHP 8.1) - and the expected version in the project .htaccess file in the root folder (PHP 8.2)... which had this code...
* NOTE: I added spaces to the code below to get it to display without editor stripping the code out - use .htaccess code as correct reference, and the below as example text to (hopefully) point you in the correct direction.

#php -- BEGIN cPanel-generated handler, do not edit
< IfModule mime module >
  AddHandler application/x-httpd-ea-php82___lsphp .php .php8 .phtml
</ IfModule >
#php -- END cPanel-generated handler, do not edit

simply, commenting out that code in .htaccess on the local server, and deleting browser cache allowed the website to be served locally - when I upload the website to the real server I will remove the comments.

#php -- BEGIN cPanel-generated handler, do not edit
#< IfModule mime_module >
#AddHandler application/x-httpd-ea-php82___lsphp .php .php8 .phtml
#</ IfModule >
#php -- END cPanel-generated handler, do not edit

You could (probably) also update Laragon to the correct PHP version for the website and see if that works..? or try hiding the .htaccess file and deleting the cache to see if the problem continues.. as it could be other conflicting settings.
Not sure if there is a better solution, but this is what worked for me..

**Reply by AWqxKAWERbXo** [reply]
Indeed, these problems are related to a bad configuration, often introduced through a custom .htaccess file inside your project folder (or even root-www folder).

**Comment by mustainal** [comment]
![image](https://github.com/user-attachments/assets/f41d65b2-623a-49ee-b24c-0b255cd3a88e)
![image](https://github.com/user-attachments/assets/d048ddce-b81d-49a5-9125-cc630b9a66ca)
